InitProject/src/test/java/MainTests.java

```diff
@@ -0,0 +1,57 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+class MainTests {
+    @Test
+    void testForEmptyList() {
+        assertEquals(0, Main.sum(new ArrayList<Integer>()));
+    }
+
+    @Test
+    void testForListOfEqualElements() {
+        assertEquals(5 * 3, Main.sum((Arrays.asList(5, 5, 5))));
+    }
+
+    @Test
+    void testCommutative() {
+        assertEquals(Main.sum(Arrays.asList(1, 2, 3)), Main.sum(Arrays.asList(3, 2, 1)));
+    }
+
+    @Test
+    void testAssociative() {
+        assertEquals(Main.sum(Arrays.asList(1, 2, 3, 4)), Main.sum(Arrays.asList(1, 2)) + Main.sum(Arrays.asList(3, 4)));
+    }
+
+    @Test
+    void testForMaxOfEmptyList() {
+        Throwable ex = assertThrows(NoSuchElementException.class, () -> {
+            Main.max(new ArrayList<Integer>());
+        });
```

Comment:
Here we can use `expression lambda` instead of `statement lambda`, and we don't really need to define a variable `ex` since we don't use it anywhere:
```java
assertThrows(NoSuchElementException.class, () -> Main.max(new ArrayList<Integer>()));
```